src/dataFiles.ts

```diff
@@ -39,65 +36,104 @@ export async function initializeDataDirs(): Promise<void> {
 	});
 	if (!dataDirs?.length) return;
 	for (const dataDir of dataDirs) {
-		watchers.set(dataDir, null);
+		modifiedPaths.set(dataDir, new Set());
```

Comment:
would it be easier for modifiedPaths to just be a Set instead of Map<string, Set>?